src/effects/Reverb.cpp

```diff
@@ -198,18 +209,6 @@ bool EffectReverb::InstanceInit(ReverbState& state, ChannelNames chanMap, bool f
 
 bool EffectReverb::ProcessFinalize()
 {
-   return InstanceDeinit(mMaster);
-}
-
-bool EffectReverb::InstanceDeinit(ReverbState& state)
-{
-   for (unsigned int i = 0; i < state.mNumChans; i++)
-   {
-      reverb_delete(&state.mP[i].reverb);
-   }
-
-   free(state.mP);
-
```

Comment:
I think you still need `state.mP.reset()` to free memory as soon as possible?

Reply:
Rebase this branch onto the other for statelessness, and it won't matter I think, because the Instance is soon destroyed when effect processing is finished.